### ISSUE TITLE
Update API (Create or Update Task) From swagger.json URL fails

### ISSUE THREAD

**Issue by MattHartz** [stephaneey/azure-apim-extension#14]
Versioned Create or Update error:
2018-10-04T17:44:02.6533208Z found api
2018-10-04T17:44:24.6744550Z ##[error]You cannot call a method on a null-valued expression.
2018-10-04T17:44:24.7326356Z ##[section]Finishing: API Management - Create/Update API 

Unversioned Create or Update error:
2018-10-04T17:27:02.7107197Z @{code=ValidationError; target=path; message=API with specified URL already exists}
2018-10-04T17:27:03.0525751Z ##[error]The remote server returned an error: (400) Bad Request.
2018-10-04T17:27:03.0897623Z ##[section]Finishing: API Management - Create/Update API 

I tried both Versioned and non-versioned, neither seem to work with Update. Create works create though!

I can see what I can do here, but i'm not too familar with powershell.

**Comment by stephaneey** [comment]
Hi,

The problem is similar to the one reported on the marketplace https://marketplace.visualstudio.com/items?itemName=stephane-eyskens.apim#qna so the solution is the same as well. APIM, whether from the portal or from Azure DevOps does not accept different facade APIs against the same backend, it only accepts different versions of the same facade API on the same backend. You probably misconfigured something either in the release pipeline, either in APIM directly using the portal.

Best Regards

**Comment by MattHartz** [comment]
Hi @stephaneey , thanks for the question response.  Yeah, it appears that I wasn't aware that the task takes in consideration the Swagger Title (at least that's what i'm guessing it uses). Because of that I had specified in the API Name section, what was already defined in my APIM.  But when I went to update it, it used what was specified in the swagger title instead, therefore making 2 APIs pointing to the same place.

Great integration you made! 👍
